src/main/java/com/lambdaschool/foundation/models/Zipcode.java

```diff
@@ -0,0 +1,74 @@
+package com.lambdaschool.foundation.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "zipcodes")
+public class Zipcode extends Auditable
+{
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long zipid;
+
+    @NotNull
+    private String code;
+
+    @ManyToOne
+    @JoinColumn(name = "cityid")
+    @NotNull
+    @JsonIgnoreProperties(value = "zipcodes")
+    private City city;
+
+    public Zipcode()
+    {
+    }
+
+    public Zipcode(
+        @NotNull String code,
+        @NotNull City city)
+    {
+        this.code = code;
+        this.city = city;
+    }
+
+    public long getZipid()
+    {
+        return zipid;
+    }
+
+    public void setZipid(long zipid)
```

Comment:
Generated id

Reply:
While we are generating the id, we still want to have the ability to change it if we want. Because the id is being generated locally on our backend, the ID isn't a parameter in any of the constructors. Being that the Id field is private we want to retain the ability to update it through the setter